### DISCUSSION TITLE
Query about Sidecar Implementation for OpenSearch Operator

### DISCUSSION THREAD

**Discussion by Yaswani341** [Opster/opensearch-k8s-operator#863] [Q&A]
Hello, I am exploring the OpenSearch Operator and was wondering if there is any ongoing work or plans to implement a sidecar pattern for certain functionalities? Specifically, I am interested in how sidecars might be used to enhance security, reloading of TLS certificates, or configuration updates dynamically. Any guidance or pointers to relevant discussions or documentation would be greatly appreciated! Thank you!

**Comment by swoehrl-mw** [comment]
Hi @Yaswani341. Currently there are no plans to either use or support sidecars. They might be utilized in the future if it is suitable for implementing specific features.
Init containers are already used in some places (preparing the volume, setting a sysctl option, initializing the keystore).
